### PR TITLE
Zero results view

### PR DESCRIPTION
**Pull request by bazelajoanna**
Add view that is displayed when there is no search result.

### CODE REVIEW

File: src/assets/styles/colors.js [hunk 1]
```diff
@@ -10,16 +10,16 @@ const darkGradientBackground =
 const lightGradientBackground =
   "linear-gradient(93.51deg, #1782DC 8.9%, #6AB7F7 94.31%)";
 
-const activeButton = "#3678E3";
+const blue100 = "#3678E3";
 
-const hoverButton = "#124CAB";
+const blue200 = "#124CAB";
+
+const blue300 = "#161C56";
```
Comment (franiakat) [inline src/assets/styles/colors.js:17]: 🙆‍♀️

File: src/components/molecules/SearchAutocomplete/index.tsx [hunk 2]
```diff
@@ -5,18 +5,23 @@ import { SearchAutocompleteProps, SUGGESTIONS } from "./interface";
...
 import * as S from "./styles";
 import { sortSuggestions } from "../../pages/ProblemIndex/utils";
 
-const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ setTag }) => {
+const SearchAutocomplete: FC<SearchAutocompleteProps> = ({
+  setTag,
+  setProblems,
+  setSearchTerm,
+  searchTerm,
+}) => {
```
Comment (evulins) [inline src/components/molecules/SearchAutocomplete/index.tsx:20]: Those propos  should also be added to test component.

 ```
 setProblems,
  setSearchTerm,
  searchTerm,
```

File: src/components/molecules/ZeroResults/index.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { ZeroResultsProps } from "./interface";
+
+import * as S from "./styles";
+
+const ZeroResults: FC<ZeroResultsProps> = ({ setTag, setSearchTerm }) => {
+  const suggestions = ["Ciągi", "Jedzenie", "Zwierzęta"];
+
+  const handleOnClick = (el: string) => {
+    setTag(el);
+    setSearchTerm(el);
```
Comment (evulins) [inline src/components/molecules/ZeroResults/index.tsx]: We  could use one style of writing **element** We can choose which one. I don't have favourite one :)

**Review by evulins — COMMENTED** [review]
Looks great! 👯

File: src/components/atoms/ResultCountLabel/index.tsx [hunk 4]
```diff
@@ -2,11 +2,14 @@ import { FC } from "react";
 
 import * as S from "./styles";
 import { ResultCountWrapperProps } from "./interface";
+import { white, primaryBackground } from "../../../assets/styles/colors";
 
 const ResultCountLabel: FC<ResultCountWrapperProps> = ({ count }) => {
   return (
     <S.ResultCountWrapper>
-      <S.ResultCountP>Wyniki wyszukiwania ({count})</S.ResultCountP>
+      <S.ResultCountP backgroundColor={count === 0 ? white : primaryBackground}>
```
Comment (franiakat) [inline src/components/atoms/ResultCountLabel/index.tsx]: I cannot see any difference in the background colour when count is 0 or non 0. 🤔

File: src/components/atoms/ResultCountLabel/styles.tsx [hunk 5]
```diff
@@ -35,13 +36,17 @@ const ResultCountWrapper = styled(Wrapper)`
   }
 `;
 
-const ResultCountP = styled(P)`
+const ResultCountP = styled(P)<ResultCountPProps>`
```
Comment (franiakat) [inline src/components/atoms/ResultCountLabel/styles.tsx]: It is not a paragraph, should be rather span I think ;)

File: src/components/molecules/ZeroResults/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  const setSearchTerm = jest.fn();
+  it("displays image", async () => {
+    testRenderer(<ZeroResults setTag={setTag} setSearchTerm={setSearchTerm} />);
+
+    expect(
+      await screen.findByLabelText("Dziewczyna patrząca przez lornetkę")
+    ).toBeInTheDocument();
+
+    let jedzenie = await screen.findByText(/Jedzenie/i);
+    fireEvent.click(jedzenie);
```
Comment (franiakat) [inline src/components/molecules/ZeroResults/index.test.tsx]:
```suggestion
    let tag = await screen.findByText(/Jedzenie/i);
    fireEvent.click(tag);
```

File: src/components/molecules/ZeroResults/index.tsx (hunk 3 above)
Comment (franiakat) [reply src/components/molecules/ZeroResults/index.tsx]: The other one would be `element`? Actually here I'd say that we have `tag` or some sort of `tagTitle` simply ;)

File: src/components/molecules/ZeroResults/index.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  return (
+    <S.ZeroResultsWrapper>
+      <S.ZeroResultsImg aria-label="Dziewczyna patrząca przez lornetkę" />
```
Comment (franiakat) [inline src/components/molecules/ZeroResults/index.tsx]: I'd skip `aria-label` and would actually hide this decoration element from screen reader ;)
```suggestion
      <S.ZeroResultsImg aria-hidden="true" />
```

File: src/components/molecules/ZeroResults/index.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      <S.ZeroResultsHeader>Brak wyników</S.ZeroResultsHeader>
+      <S.ZeroResultsP>Spróbuj wyszukać:</S.ZeroResultsP>
```
Comment (franiakat) [inline src/components/molecules/ZeroResults/index.tsx]: It's not a paragraph, a span or tiny heading ;)

File: src/components/molecules/ZeroResults/index.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import { FC } from "react";
+import { ZeroResultsProps } from "./interface";
+
+import * as S from "./styles";
+
+const ZeroResults: FC<ZeroResultsProps> = ({ setTag, setSearchTerm }) => {
+  const suggestions = ["Ciągi", "Jedzenie", "Zwierzęta"];
```
Comment (franiakat) [inline src/components/molecules/ZeroResults/index.tsx]:
```suggestion
  const suggestedTags = ["Ciągi", "Jedzenie", "Zwierzęta"];
```

File: src/components/molecules/ZeroResults/index.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    setSearchTerm(el);
+  };
+
+  return (
+    <S.ZeroResultsWrapper>
+      <S.ZeroResultsImg aria-label="Dziewczyna patrząca przez lornetkę" />
+      <S.ZeroResultsHeader>Brak wyników</S.ZeroResultsHeader>
+      <S.ZeroResultsP>Spróbuj wyszukać:</S.ZeroResultsP>
+      <S.ZeroResultsSuggestions>
+        {suggestions.map((suggestion, id) => (
```
Comment (franiakat) [inline src/components/molecules/ZeroResults/index.tsx]:
```suggestion
        {suggestedTag.map((tag, id) => (
```

File: src/components/molecules/ZeroResults/index.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    setTag(el);
+    setSearchTerm(el);
+  };
+
+  return (
+    <S.ZeroResultsWrapper>
+      <S.ZeroResultsImg aria-label="Dziewczyna patrząca przez lornetkę" />
+      <S.ZeroResultsHeader>Brak wyników</S.ZeroResultsHeader>
+      <S.ZeroResultsP>Spróbuj wyszukać:</S.ZeroResultsP>
+      <S.ZeroResultsSuggestions>
```
Comment (franiakat) [inline src/components/molecules/ZeroResults/index.tsx]:
```suggestion
      <S.ZeroResultsTagSuggestions>
```

File: src/components/pages/ProblemIndex/index.test.tsx [hunk 12]
```diff
@@ -35,4 +35,20 @@ describe("<ProblemIndex/>", () => {
       await screen.findByText("Wyniki wyszukiwania (2)")
     ).toBeInTheDocument();
   });
+
+  xit("displays zero results view when result count = 0", async () => {
```
Comment (franiakat) [inline src/components/pages/ProblemIndex/index.test.tsx:39]: 😓

**Review by franiakat — CHANGES_REQUESTED** [review]
Please see a few suggestions :))

File: src/components/atoms/ResultCountLabel/index.tsx (hunk 4 above)
Comment (bazelajoanna) [reply src/components/atoms/ResultCountLabel/index.tsx]: According to design, background colour changes only on mobile. For tablets and wider screens there is no difference.

File: src/components/molecules/ZeroResults/index.tsx (hunk 3 above)
Comment (bazelajoanna) [reply src/components/molecules/ZeroResults/index.tsx]: Ok, I've changed `el` to `tag`.

File: src/components/atoms/ResultCountLabel/styles.tsx (hunk 5 above)
Comment (bazelajoanna) [reply src/components/atoms/ResultCountLabel/styles.tsx]: Good point, I've changed it :)

File: src/components/molecules/ZeroResults/index.tsx (hunk 8 above)
Comment (bazelajoanna) [reply src/components/molecules/ZeroResults/index.tsx]: Right, I've changed it :)

File: src/components/atoms/ResultCountLabel/index.tsx (hunk 4 above)
Comment (franiakat) [reply src/components/atoms/ResultCountLabel/index.tsx]: Yes, but why should it depend on the count? If I understand it correctly it should be connected with the viewport size (mobile).

Comment (franiakat) [reply src/components/atoms/ResultCountLabel/index.tsx]: Nvm, I've misread the design. ;)

File: src/components/molecules/ZeroResults/index.tsx (hunk 7 above)
Comment (franiakat) [reply src/components/molecules/ZeroResults/index.tsx]: @bazelajoanna

Comment (bazelajoanna) [reply src/components/molecules/ZeroResults/index.tsx]: Done ;)

**Review by franiakat — APPROVED** [review]
sqash

**Pull request merged**